packages/ra-ui-materialui/src/input/ResettableTextField.js

```diff
@@ -99,37 +101,50 @@ function ResettableTextField({
-                                [visibleClearButton]:
-                                    clearAlwaysVisible || showClear,
-                            })}
-                            aria-label={translate(
-                                'ra.action.clear_input_value'
-                            )}
-                            title={translate('ra.action.clear_input_value')}
-                            disableRipple
-                            onClick={handleClickClearButton}
-                            onMouseDown={handleMouseDownClearButton}
-                            disabled={disabled}
+                endAdornment:
+                    resettable && value ? (
+                        <InputAdornment
+                            position="end"
+                            classes={{
+                                root: props.select ? selectAdornment : null,
+                            }}
                         >
-                            <ClearIcon
-                                className={classNames(clearIcon, {
-                                    [visibleClearIcon]:
+                            <IconButton
+                                className={classNames(clearButton, {
+                                    [visibleClearButton]:
                                         clearAlwaysVisible || showClear,
                                 })}
-                            />
-                        </IconButton>
-                    </InputAdornment>
-                ),
-                ...InputProps,
+                                aria-label={translate(
+                                    'ra.action.clear_input_value'
+                                )}
+                                title={translate('ra.action.clear_input_value')}
+                                disableRipple
+                                onClick={handleClickClearButton}
+                                onMouseDown={handleMouseDownClearButton}
+                                disabled={disabled}
+                            >
+                                <ClearIcon
+                                    className={classNames(clearIcon, {
+                                        [visibleClearIcon]:
+                                            clearAlwaysVisible || showClear,
+                                    })}
+                                />
+                            </IconButton>
+                        </InputAdornment>
+                    ) : (
+                        endAdornment ||
+                        (resettable && (
+                            <InputAdornment
+                                position="end"
+                                classes={{
+                                    root: props.select ? selectAdornment : null,
+                                }}
+                            >
+                                <IconButton disabled />
```

Comment:
you're showing the reset button even though there is no content? I don't like that idea, I think it adds useless visual information.

Reply:
It doesn't show anything, maintaining a `IconButton` prevents the input with from changing when `resettable` is  true and there is no value in the input.
If I'm not mistaken this was your code, I just added the disabled prop to avoid user interaction and the check to see if `resettable` is true.

Reply:
I'm looking into it, you are right, plus on blur, no addorment is showing.
I'll pul something soon.

Reply:
Done.
I don't know if is the cleanest way to  do it but it works.
Let me know.